### ISSUE TITLE
Overwrite default root location

### ISSUE THREAD

**Issue by elboletaire** [HanXHX/ansible-nginx#4]
Continuing from the resulting file on #3 (please, read that issue first), with that file I can access my transmission from `/transmission/web/`. Now I want to access transmission from the root path.

I know that I could do some tricks with rewrite rules*, but for the sake of simplicity I'd like to redirect my calls to the root path to `/transmission/web/` using this location:

``` nginx
location / {
  return 302 /transmission/web/;
}
```

But I cannot do this using the ansible role because the root location is determined by the template being used.

This example is for transmission but I've found other cases before where rewriting the root location would be very useful (or at least allowing us to add custom params to it, before the `try_files`).

<sub>* btw, I don't know where should I configure these rewrite rules if I cannot add anything to the root location.</sub>

**Comment by elboletaire** [comment]
I've fixed the transmission issue using the proxy template, but it's still interesting to overwrite or at least add parameters to the root location for other templates such as `_base` or `_php`.

**Comment by HanXHX** [comment]
Hi @elboletaire 

Did you try to use "root" as vhost argument?

Can you paste your vars?

Thank-you

**Comment by elboletaire** [comment]
Yes, I use root on all my vhosts. I cannot currently give you an example generated with your role because, as said, my current sites and proxies work as expected. But I'll paste you an example (symfony app) that I don't know how to reproduce with your role without rewriting the root location:

``` nginx
server {
    listen   80;
    server_name  davantis.caprica.cir;
    root   /var/www/davantis/web;

    location / {
        # try to serve file directly, fallback to app.php
        try_files $uri /app.php$is_args$args;
    }
    # DEV
    # This rule should only be placed on your development environment
    # In production, don't include this and don't deploy app_dev.php or config.php
    location ~ ^/(app_dev|config)\.php(/|$) {
        fastcgi_pass 127.0.0.1:9000;
        fastcgi_split_path_info ^(.+\.php)(/.*)$;
        include fastcgi_params;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_param HTTPS off;
    }
    # PROD
    location ~ ^/app\.php(/|$) {
        fastcgi_pass 127.0.0.1:9000;
        fastcgi_split_path_info ^(.+\.php)(/.*)$;
        include fastcgi_params;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_param HTTPS off;
    }
    # PROD
    location ~ ^/app\.php(/|$) {
        fastcgi_pass 127.0.0.1:9000;
        fastcgi_split_path_info ^(.+\.php)(/.*)$;
        include fastcgi_params;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_param HTTPS off;
        # Prevents URIs that include the front controller. This will 404:
        # http://domain.tld/app.php/some-path
        # Remove the internal directive to allow URIs like this
        internal;
    }

    error_log /var/log/nginx/davantis_error.log;
    access_log /var/log/nginx/davantis_access.log;
}

```

**Comment by HanXHX** [comment]
Can you paste me expected vhost? I don't understand your need.

**Comment by elboletaire** [comment]
How can I overwrite the `try_files` block in the `_php` template so it is like the one in the file above?

I need this location:

``` nginx
location / {
      # try to serve file directly, fallback to app.php
      try_files $uri /app.php$is_args$args;
}
```

But I don't know how should I do it with the given templates.

**Comment by HanXHX** [comment]
Hi @elboletaire, i'm working on this.

**Comment by HanXHX** [comment]
Hi @elboletaire,

I did this job on a remote branch with PR: https://github.com/HanXHX/ansible-nginx/pull/6

Can you check everything is ok for you? I'll merge and release if it's ok.

Cheers!

**Comment by elboletaire** [comment]
You allow to redefine the `try_files`, but what if I want to add something before it? I think that a way to entirely replace the `/` location would be awesome.

How would you approach something like this?

``` nginx
location / { 
    if ($host !~ caprica.cir$) {
        return 403;
    }   
    try_files $uri $uri/index.html $uri.html $uri.php @markdown;
}
```

**Comment by HanXHX** [comment]
"If" is evil! You should read: https://www.nginx.com/resources/wiki/start/topics/depth/ifisevil/

**Comment by elboletaire** [comment]
Didn't know that!

Ok then you can close the issue but let me know if you know an alternate way of doing what I do there (which works as expected though).

**Comment by HanXHX** [comment]
Hi @elboletaire 

I merged #6. If you need anything else, please open another issue. I close this one.

Cheers!

**Comment by elboletaire** [comment]
Nice, thanks!
